service/src/models/archaeologist.ts

```diff
@@ -67,6 +69,35 @@ export class Archaeologist {
       } catch (err) {
         console.log("problem with pipe", err)
       }
+    });
+
+    this.node.handle(['/validate-arweave'], async ({ stream }) => {
+      try {
+        await pipe(stream, async (source) => {
+          for await (const data of source) {
+            const jsonData = JSON.parse(new TextDecoder().decode(data));
+
+            const txId = jsonData.arweaveTxId;
+            const unencryptedShardHash = jsonData.unencryptedShardHash;
+
+            const wallet = new ethers.Wallet(process.env.ENCRYPTION_PRIVATE_KEY!);
+
+            const isValidShard = await fetchAndValidateArweaveShard(txId, unencryptedShardHash, wallet.publicKey);
+
+            if (isValidShard) {
+              const msg = ethers.utils.solidityPack(['string', 'string'], [txId, unencryptedShardHash])
+              const signature = await wallet.signMessage(msg);
+
+              const toBrowser = JSON.stringify(signature);
+              this.publish("env-config", `signature ${toBrowser}`);
+            } else {
+              this.publish("env-config", `signature ${null}`);
+            }
```

Comment:
Should this use the `archLogger` and throw an error? 

We may want to setup typed Errors (enum or object) that could be copied over to the web app. These would help inform the webapp of exactly what is going wrong when something outside of the "happy path" happens. This can be handled in a followup PR.